nixos/modules/programs/lomiri.nix

```diff
@@ -0,0 +1,160 @@
+        lomiri-session # Wrappers to properly launch the session
+        lomiri
+        qtmir # not having its desktop file for Xwayland available causes any X11 application to crash the session
+        lomiri-system-settings
+        morph-browser
+        lomiri-terminal-app
+
+        libusermetrics
+
+        # TODO OSK does not work yet
+        # lomiri-keyboard is a plugin for maliit-framework, which still seems incapable of loading any plugins at all.
+        # maliit-framework's maliit-server hardcodes a plugin name & location to load, and needs the plugin's glib schema.
+        # It (maliit-server or the plugin?) also has a hardcoded path to hunspell dictionaries for text prediction
+        #maliit-framework
+        #lomiri-keyboard
+
+        # Required/Expected user services
+        lomiri-thumbnailer
+        lomiri-url-dispatcher
+        lomiri-download-manager
+        lomiri-schemas # exposes some required dbus interfaces
+        hfd-service
+        history-service
+        telephony-service
+        content-hub
+        mediascanner2 # possibly needs to be kicked off by graphical-session.target
+
+        lomiri-sounds
+        lomiri-wallpapers
+        suru-icon-theme
+      ]);
+    };
+
+    # Required/Expected system services
+    systemd.packages = with pkgs.lomiri; [
+      hfd-service
+      lomiri-download-manager
+    ];
+
+    services.dbus.packages = with pkgs.lomiri; [
+      hfd-service
+      libusermetrics
+      lomiri-download-manager
+    ];
+
+    fonts.packages = with pkgs; [
+      # Applications tend to default to Ubuntu font
+      ubuntu_font_family
+    ];
+
+    # Copy-pasted
+    # TODO are all of these needed? just nice-have's? convenience?
+    hardware.opengl.enable = lib.mkDefault true;
+    fonts.enableDefaultPackages = lib.mkDefault true;
+    programs.dconf.enable = lib.mkDefault true;
+    programs.xwayland.enable = lib.mkDefault true;
+
+    services.accounts-daemon.enable = true;
+
+    services.ayatana-indicators = {
+      enable = true;
+      packages = (with pkgs; [
+        ayatana-indicator-datetime
+        ayatana-indicator-messages
+        ayatana-indicator-session
+      ]) ++ (with pkgs.lomiri; [
+        telephony-service
+      ]);
+    };
+
+    services.udisks2.enable = true;
+    services.upower.enable = true;
+    services.geoclue2.enable = true;
+    services.printing.enable = true;
+
+    services.gnome.evolution-data-server = {
+      enable = true;
+      plugins = with pkgs; [
+        #address-book-service
+      ];
+    };
+
+    services.telepathy.enable = true;
+
+    services.xserver.displayManager.lightdm.enable = lib.mkDefault true;
+    services.xserver.displayManager.lightdm.greeters.lomiri.enable = lib.mkDefault true;
+    services.xserver.displayManager.defaultSession = lib.mkDefault "lomiri";
+    services.xserver.displayManager.sessionPackages = with pkgs.lomiri; [ lomiri-session ];
+    services.xserver.updateDbusEnvironment = true;
```

Comment:
`updateDbusEnvironment` don't work for session configured by `services.xserver.displayManager.sessionPackages`, 

It add  `${lib.getBin pkgs.dbus}/bin/dbus-update-activation-environment --systemd --all ` to `xsession` script, which only be generated when  `services.xserver.displayManager.session` was used

https://github.com/NixOS/nixpkgs/issues/209847

Reply:
Will drop it from here.